### ISSUE TITLE
follow pytorch_mitsuba_interoperability tutorials, my own python scipt's render_img.requires_frad is False.

### ISSUE THREAD

**Issue by anewusername77** [mitsuba-renderer/mitsuba3#459]
## Summary

Big thanks to mitsuba team's excellent work! 
I'm trying to use this differetiable renderer in network training, with `pytorch optim pipeline` to optimize mitsuba params like  `specular` in principled bsdf and `vertex_positions` in mesh.

## System configuration

System information:

  OS: Ubuntu 20.04.4 LTS
  CPU: Intel(R) Xeon(R) Platinum 8276L CPU @ 2.20GHz
  GPU: Tesla V100-PCIE-32GB
Tesla V100-PCIE-32GB
  Python: 3.8.13 (default, Mar 28 2022, 11:38:47) [GCC 7.5.0]
  NVidia driver: 470.129.06
  LLVM: 12.0.0

  Dr.Jit: 0.3.2
  Mitsuba: 3.1.1
     Is custom build? False
     Compiled with: GNU 10.2.1
     Variants:
        scalar_rgb
        scalar_spectral
        cuda_ad_rgb
        llvm_ad_rg

## Description
read this tutorial already: https://mitsuba.readthedocs.io/en/latest/src/inverse_rendering/pytorch_mitsuba_interoperability.html

my question is:
I noticed that `specular`,  `roughness.value` and `base_color.data` belong to `mi.Float`, `mi.Float` and  `mi.TensorXf` respectively, are they differentiable? since float in python is scalar. 
I want to use  `l1loss(rendered_img, gt_img)` to update network's params like that in the tutorial, so I need the gradient of them can be calculated, while other params like `spec_tint` need to be frozen during optimizing.

code:
```python
def setupBSDF(self, spec:torch.Tensor, roughness: torch.Tensor, albedo_tex:torch.Tensor):
    tmp = mi.Bitmap('diffuseMap_0.png')
    bsdf = mi.load_dict({
            'type': 'principled',
            'base_color': {
                'type': 'bitmap',
                'bitmap': tmp,
            },
            'specular':0.2,
            'roughness': 0.5,
        })
    params = mi.traverse(bsdf)
    params['base_color.data'] = albedo_tex 
    params['specular'] = spec
    params['roughness.value'] = roughness
    params.update()
    return bsdf
....
@dr.wrap_ad(source='torch', target='drjit')
# mesh includes bsdf
def renderimg(mesh, envmap:torch.Tensor, vertices:torch.Tensor, faces:torch.Tensor, 
normals:torch.Tensor, sensor):  
  scene = mi.load_dict({
                "type": "scene",
                "integrator": {
                    "type": "path",
                    'max_depth': 8,
                    'hide_emitters': True},
                "light": {
                    'filename': 'tmpenv.png',
                    'type': 'envmap'},
                'bfm': mesh
            })
        
        params_scene = mi.traverse(scene)
        params_scene['light.data'] = envmap
        
        params_scene['bfm.vertex_positions'] = dr.ravel(vertices)
        params_scene['bfm.vertex_normals'] = dr.ravel(normals)
        params_scene['bfm.faces'] = dr.ravel(indices)

  params_scene.update()
  return mi.render(scene, sensor=sensor, spp=self.samples)
```
but `rendered_img.requires_grad` is `False`

**Comment by anewusername77** [comment]
mesh code is like:
```python
def setupMesh(vertices:torch.Tensor, faces:torch.Tensor):
    vertices = vertices.cpu().numpy()  # n, 3
    faces = faces.cpu().numpy()          # n, 3
    vertex_pos = mi.Vector3f(mi.Float(vertices[:,0]),
                                mi.Float(vertices[:,1]),
                                mi.Float(vertices[:,2]))
    face = mi.Vector3u(mi.UInt(faces[:,0]),
                          mi.UInt(faces[:,1]),
                          mi.UInt(faces[:,2]))
   mesh = mi.Mesh(
            "wavydisk",
            vertex_count=10,
            face_count=30,
            has_vertex_normals=True
   )
   mesh_params = mi.traverse(mesh)
   mesh_params["vertex_positions"] = dr.ravel(vertex_pos)
   mesh_params["faces"] = dr.ravel(face)
   mesh_params.update()
```
I tried `mesh_params["vertex_positions"] = torch.Tensor(vertices)` but it raises error. 
`mesh_params["vertex_positions"] = mi.TensorXf(torch.Tensor(vertices))` runs fine.

**Comment by anewusername77** [comment]
new question， is it possible to optimize `focal_length` in sensor?

**Comment by anewusername77** [comment]
I can't change the scene by:
```python
params_scene['bfm.bsdf.specular'] = spec               # type(spec)== torch.Tensor
```
how can i make specular differentiable and able to be optimized?

**Comment by anewusername77** [comment]
i tried using ```params_scene['bfm.bsdf.specular'] = dr.ravel(specular) ```, and ```optimizer = torch.optim.Adam([{'params': specular, 'lr': 0.001}])``` . but although `specular.requires_grad=True`, it didn't update at all.

**Comment by anewusername77** [comment]
incase anyone else has this question, solve it by `img = mi.render(scene, params, seed, spp)`

**Comment by njroussel** [comment]
To make it a bit more clear to anyone that comes across this thread.

When using differentiating through the `mi.render` function, it is crucial to pass the scene params as an arugment too, as you can see in @anewusername77  previous message.

Glad you solved this :)
